### PR TITLE
Adding constraints to computer example

### PR DESCRIPTION
**Pull request by david-waltermire-nist**
# Committer Notes

This PR adds constraints to the notional computer example.

These constraints can support discussion around #411 and work in PR #413.

### All Submissions:

- [ ] Have you followed the guidelines in our [Contributing](https://github.com/usnistgov/metaschema/blob/master/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/usnistgov/metaschema/pulls) for the same update/change?
- [ ] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [ ] Do all automated CI/CD checks pass?

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you included examples of how to use your new feature(s)?
- [ ] Have you updated all website](https://pages.nist.gov/metaschema) and readme documentation affected by the changes you made? Changes to the website can be made in the website/content directory of your branch.

### CODE REVIEW

File: website/content/specification/computer-example.xml [hunk 1]
```diff
@@ -149,6 +178,19 @@
                 <description>Ensures that each USB device has a unique UUID.</description>
                 <key-field target="@uuid"/>
             </is-unique>
+            <allowed-values target="(motherboard|motherboard/cpu|motherboard/memory|motherboard/expansion-card|usb-device)/prop[not(@ns) or @ns=('http://csrc.nist.gov/ns/metaschema/computer/example')]/@name">
+                <enum value="model">CPU model</enum>
+            </allowed-values>
+            <has-cardinality level="WARNING" target="(motherboard|motherboard/cpu|motherboard/memory|motherboard/expansion-card|usb-device)/prop[not(@ns) or @ns=('http://csrc.nist.gov/ns/metaschema/computer/example')]" min-occurs="1">
+                <remarks>
+                    <p>Warn if a model is not provided.</p>
```
Comment (aj-stein-nist) [inline website/content/specification/computer-example.xml:186]: Minor, non-blocking: maybe we should align with the style of the remark below? It sounds like the expectation is to make the remark seem like a user-facing message, even if it is not one.

```suggestion
                    <p>It is invalid to not provide a CPU model.</p>
```

File: website/content/specification/computer-example.xml [hunk 2]
```diff
@@ -164,8 +206,59 @@
...
+            <formal-name>Property Universally Unique Identifier</formal-name>
+            <!-- identifier declaration -->
+            <description>A unique identifier for a property.</description>
+        </define-flag>
+        <define-flag name="ns" as-type="uri" default="http://csrc.nist.gov/ns/metaschema/computer/example">
+            <formal-name>Property Namespace</formal-name>
+            <description>A namespace qualifying the property's name. This allows different organizations to associate distinct semantics with the same name.</description>
+            <remarks>
+                <p>This value must be an <a href="https://datatracker.ietf.org/doc/html/rfc3986#section-4.3">absolute URI</a> that serves as a naming system identifier.</p>
+                <p>When a <code>ns</code> is not provided, its value should be assumed to be <code>http://csrc.nist.gov/ns/oscal</code> and the name should be a name defined by the associated OSCAL model.</p>
```
Comment (aj-stein-nist) [inline website/content/specification/computer-example.xml:227]: Blocking: I assume we want to clean up OSCAL references.

```suggestion
                <p>When a <code>ns</code> is not provided, its value should be assumed to be <code>http://example.com/ns/computer</code> and the name should be a name defined by the associated computer model.</p>
```

File: website/content/specification/computer-example.xml [hunk 3]
```diff
@@ -164,8 +206,59 @@
...
+            <description>An identifier for relating distinct sets of properties.</description>
+            <remarks>
+                <p>Different sets of properties may relate to separate contexts. Declare a group on a property to associate it with one or more other properties in a given context.</p>
+            </remarks>
+        </define-flag>
+        <model>
+            <field ref="remarks"/>
+        </model>
+        <constraint>
+            <allowed-values target=".[not(@ns) or @ns=('http://csrc.nist.gov/ns/metaschema/computer/example')]/@name">
```
Comment (aj-stein-nist) [inline website/content/specification/computer-example.xml:254]: Blocking: should we make this align with the schema URI (or change `namespace` but I can't suggest that with the GitHub PR suggestion interface).

```suggestion
            <allowed-values target=".[not(@ns) or @ns=('http://example.com/ns/computer')]/@name">
```

File: website/content/specification/computer-example.xml [hunk 4]
```diff
@@ -164,8 +206,59 @@
...
+                <enum value="marking">A label or descriptor that is tied to a sensitivity or classification marking system. An optional class can be used to define the specific marking system used for the associated value.</enum>
+            </allowed-values>
+        </constraint>
+        <remarks>
+            <p>Properties permit the deployment and management of arbitrary controlled values, within OSCAL objects. A property can be included for any purpose useful to an application or implementation. Typically, properties will be used to sort, filter, select, order, and arrange OSCAL content objects, to relate OSCAL objects to one another, or to associate an OSCAL object to class hierarchies, taxonomies, or external authorities. Thus, the lexical composition of properties may be constrained by external processes to ensure consistency.</p>
+            <p>Property allows for associated remarks that describe why the specific property value was applied to the containing object, or the significance of the value in the context of the containing object.</p>
```
Comment (aj-stein-nist) [inline website/content/specification/computer-example.xml:260]: Blocking: should we remove OSCAL references?

```suggestion
            <p>Properties permit the deployment and management of arbitrary controlled values, within computer objects. A property can be included for any purpose useful to an application or implementation. Typically, properties will be used to sort, filter, select, order, and arrange computer content objects, to relate computer objects to one another, or to associate an computer object to class hierarchies, taxonomies, or external authorities. Thus, the lexical composition of properties may be constrained by external processes to ensure consistency.</p>
            <p>Property allows for associated remarks that describe why the specific property value was applied to the containing object, or the significance of the value in the context of the containing object.</p>
```

File: website/content/specification/computer-example.xml [hunk 5]
```diff
@@ -1,5 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<METASCHEMA xmlns="http://csrc.nist.gov/ns/oscal/metaschema/1.0">
+<METASCHEMA xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+ xsi:schemaLocation="http://csrc.nist.gov/ns/oscal/metaschema/1.0 ../../../../metaschema-java/core/metaschema/schema/xml/metaschema.xsd"
```
Comment (aj-stein-nist) [inline website/content/specification/computer-example.xml:3]: Question: is this the actual xsi:schemaLocation with the `/oscal/` in it? This made me confused.

File: website/content/specification/computer-example.xml [hunk 6]
```diff
@@ -1,5 +1,7 @@
...
+ xmlns="http://csrc.nist.gov/ns/oscal/metaschema/1.0">
```
Comment (aj-stein-nist) [inline website/content/specification/computer-example.xml:4]: Question: ditto here?

File: website/content/specification/computer-example.xml [hunk 7]
```diff
@@ -64,11 +72,22 @@
...
+                                        <enum value="arm">arm</enum>
+                                    </allowed-values>
+                                </constraint>
                             </define-field>
                             <define-field name="speed" min-occurs="1">
                                 <formal-name>CPU Speed</formal-name>
                                 <description>The clock speed of the CPU in megahertz or gigahertz.</description>
+                                <constraint>
+                                    <matches target="." regex="\d+(?:\.\d+)?(?:MHz|GHz)"></matches>
+                                </constraint>
```
Comment (aj-stein-nist) [inline website/content/specification/computer-example.xml:88]: Minor, non-blocking: I like this but I thought I had made this originally a `positiveInteger` datatype and was planning on accompanying that with something like a `speed-unit`, but I can see that instructive to me but a coding/modeling smell (I actually have never formulated this "style" question in the Metaschema community). Just a heads up, no suggested change here, feel free to resolve.

**Review by aj-stein-nist — CHANGES_REQUESTED** [review]

File: website/content/specification/computer-example.xml (hunk 1 above)
Comment (david-waltermire-nist) [reply website/content/specification/computer-example.xml:186]: Ignoring this based on your verbal feedback.

File: website/content/specification/computer-example.xml (hunk 5 above)
Comment (david-waltermire-nist) [reply website/content/specification/computer-example.xml:3]: Yes indeed!

File: website/content/specification/computer-example.xml (hunk 7 above)
Comment (david-waltermire-nist) [reply website/content/specification/computer-example.xml:88]: I setup this constraint based on the current model. We could always break this field apart into value and unit data elements. We should probably do this in a different PR.

**Review by aj-stein-nist — APPROVED** [review]
I know this is draft, but I am good with this as-is after addressing my feedback.

**Review by wendellpiez — APPROVED** [review]
Fantastic, thanks!

**Pull request merged**